src/mme-app/utils/mmeNasUtils.cpp

```diff
@@ -669,6 +669,98 @@ void MmeNasUtils::decode_tau_req(unsigned char *msg, int &nas_msg_len,
     }
 }
 
+void MmeNasUtils::decode_act_ded_br_ctxt_acpt(unsigned char *msg, int& nas_msg_len, struct nasPDU *nas)
+{
+	uint8_t elem_id = msg[0];
+        uint8_t index = 0;
+        uint8_t datalen = 0;
+        unsigned char *msg_end = msg + (nas_msg_len - ESM_HEADER_LEN);
+
+	nas->elements_len = 1;
+        nas->elements = (nas_pdu_elements *) calloc(sizeof(nas_pdu_elements), 1);
+
+        bool status = true;
+        while (status && (msg < msg_end))
+	{
+            elem_id = msg[0];
+	    switch (elem_id)
+            {
+		case NAS_IE_TYPE_PCO:
+        	{
+            		nas->elements[index].msgType = NAS_IE_TYPE_PCO;
+	   	 	msg++; //Skipping Element ID
+            		memcpy(&nas->elements[index].pduElement.pco_opt.pco_length, msg, 1);
+            		msg++;
+            		memcpy(&nas->elements[index].pduElement.pco_opt.pco_options[0], msg,
+                                nas->elements[index].pduElement.pco_opt.pco_length);
+			msg += nas->elements[index].pduElement.pco_opt.pco_length;
+			index++;
+			break;
+        	}
+		case NAS_IE_TYPE_NBIFOM_CONTAINER:
+		case NAS_IE_TYPE_EXT_PCO:
+		{
+			msg++; //skipping Element ID
+            		datalen = msg[0];
+            		msg++;
+            		msg += datalen;
+            		break;
+		}
+		default:
+        	{
+            		log_msg(LOG_WARNING, "NAS ACT_DED_BR_CTXT_ACPT - Unhandled IE %x\n", elem_id);
+
+            		// set status to false to break out of while
```

Comment:
We should skip unknown elements in NAS message payload. NAS payload comes from UE and we may have variety of UE equipments which various release support. So this piece of code need to be robust.

Reply:
We have handled all IEs as per 23.401 v15.6.0

The length is not a mandatory field in a few of the NAS IEs, so we wont be able to fetch the length and skip the bytes unless we know which exact IE we are currently decoding.

If we keep incrementing the msg pointer (msg++) for default condition, we may encounter a byte that has same IE value as any of the valid IEs that we have handled, and may end up overwriting them. That's why we are stopping the decode once when we have encountered unhandled IE.